exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSystemPlugin.java

```diff
@@ -74,6 +74,7 @@ public FileSystemPlugin(FileSystemConfig config, DrillbitContext context, String
           fsConf.set(s, config.config.get(s));
         }
       }
+      fsConf.set("fs.default.name", config.connection);
```

Comment:
Please use `FileSystem.FS_DEFAULT_NAME_KEY`. It seems that `"fs.default.name"` is deprecated.

But, note, that is exactly what the next line of code does. So, perhaps there is more going on here than a first read suggests? Or, are we confusing ourselves about the meaning of the two properties?

Reply:
Doing a quick google it looks like you are correct. **fs.default.name** is deprecated and we should use **fs.defaultFS** instead https://stackoverflow.com/questions/30480847/difference-between-fs-defaultfs-and-fs-default-name?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa. Also hortonworks dropped **fs.default.name** from Ambari a long time ago so we should be safe https://issues.apache.org/jira/browse/AMBARI-2789 .

Reply:
Okay I think I have a better handle on this now. The original issue was that Drill's hive storage plugin had a configuration option of fs.default.name = file:// . Somehow when a hive table was dropped and then recreated with a ctas statement in drill, the CTAS statement picked up the fs.default.name configuration from the hive storage plugin and passed that on to DrillFileSystem. And apparently if both fs.default.name and fd.defaultFS are present with different values the value for fs.default.name wins even though it is deprecated. So the CTAS statement would end up creating the table on a drill node's local filesystem.

I believe the crux of this PR is to force "fs.default.name" to have to correct value in the event that a different value is defined in the HiveStorage plugin.

With that said, there are several questions. 

 1. How the heck does a property in the HiveStoragePlugin make it's way into the FileSystem configuration? I spent a good amount of time looking at the code and for the life of me I can't figure that out.
 2. The follow up to (1) is do we actually want that behavior? We can force fs.default.name to have the right value but what about other properties we might suck in from a HiveStoragePlugin configuration?
 3. If we don't want this behavior what would be the real fix?

In the face of all this ambiguity I think we should move forward with a minimal PR that forces fs.default.name to be correct now. We can have a follow up Jira that actually fixes the underlying problem of sucking in stray configs down the road if someone complains about it.